### PR TITLE
Upgrade BWC version to 2.18

### PR DESCRIPTION
**Pull request by martin-gaievski**
### Description
Bump one version up for BWC tests workflow after 2.17 got released and next 2.x has updated to 2.18.
For 2.x it will be a separate PR as these changes cannot be backported "as is". 

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [X] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/neural-search/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/backwards_compatibility_tests_workflow.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       matrix:
         java: [ 21 ]
         os: [ubuntu-latest,windows-latest]
-        bwc_version : ["2.9.0","2.10.0","2.11.0","2.12.0","2.13.0","2.14.0","2.15.0","2.16.0","2.17.0-SNAPSHOT"]
+        bwc_version : ["2.9.0","2.10.0","2.11.0","2.12.0","2.13.0","2.14.0","2.15.0","2.16.0","2.17.0", "2.18.0-SNAPSHOT"]
```
Comment (vibrantvarun) [inline .github/workflows/backwards_compatibility_tests_workflow.yml:18]: nit: 
```suggestion
        bwc_version : ["2.9.0","2.10.0","2.11.0","2.12.0","2.13.0","2.14.0","2.15.0","2.16.0","2.17.0","2.18.0-SNAPSHOT"]
```

File: gradle.properties [hunk 2]
```diff
@@ -7,7 +7,7 @@
 # https://github.com/opensearch-project/OpenSearch/blob/main/libs/core/src/main/java/org/opensearch/Version.java .
 # Wired compatibility of OpenSearch works like 3.x version is compatible with 2.(latest-major) version.
 # Therefore, to run rolling-upgrade BWC Test on local machine the BWC version here should be set 2.(latest-major).
-systemProp.bwc.version=2.17.0-SNAPSHOT
+systemProp.bwc.version=2.18.0-SNAPSHOT
 systemProp.bwc.bundle.version=2.17.0
```
Comment (vibrantvarun) [inline gradle.properties:11]:
```suggestion
systemProp.bwc.bundle.version=2.18.0
```

File: .github/workflows/backwards_compatibility_tests_workflow.yml (hunk 1 above)
Comment (martin-gaievski) [reply .github/workflows/backwards_compatibility_tests_workflow.yml]: ack

File: gradle.properties (hunk 2 above)
Comment (martin-gaievski) [reply gradle.properties]: ack

**Review by vibrantvarun — APPROVED** [review]

**Review by junqiu-lei — APPROVED** [review]

**Pull request merged**